crates/core/tedge_api/src/lib.rs

```diff
@@ -32,6 +32,7 @@ pub use software::*;
 
 pub const DEFAULT_DATA_PATH: &str = "/var/tedge";
 pub const DEFAULT_FILE_TRANSFER_DIR_NAME: &str = "file-transfer";
+pub const DEFAULT_FILE_CACHE_DIR_NAME: &str = "cache";
```

Comment:
I would really avoid to add yet another `pub const`.  

These constants are too low level. Look at the current usages for `DEFAULT_FILE_TRANSFER_DIR_NAME`. All of these are starting with a `data_dir.join()`. i.e. doing the same as `file_transfer_dir_as_pathbuf()` but with some possibly buggy `data_dir`.

What we need are methods providing paths ready to be used.